silkworm/capi/silkworm_test.cpp

```diff
@@ -965,6 +970,132 @@ TEST_CASE_METHOD(CApiTest, "CAPI silkworm_add_snapshot", "[silkworm][capi]") {
+    using snapshots::Schema;
+    using namespace snapshots;
+    constexpr uint32_t kZeroSalt{0};
+
+    const bloom_filter::BloomFilterKeyHasher bloom_key_hasher{kZeroSalt};
+
+    const snapshot_test::SampleAccountsDomainSegmentFile accounts_kv_file{tmp_dir.path()};
+    const auto& accounts_kv_path = accounts_kv_file.path();
+    segment::KVSegmentFileReader accounts_segment{accounts_kv_path, seg::CompressionKind::kAll};
+    const snapshot_test::SampleAccountsDomainKVEIFile accounts_kvei_file{tmp_dir.path()};
+    const auto& accounts_kvei_path = accounts_kvei_file.path();
+    bloom_filter::BloomFilter account_existence_index{accounts_kvei_path.path(), bloom_key_hasher};
+    const snapshot_test::SampleAccountsDomainBTFile accounts_bt_file{tmp_dir.path()};
+    const auto& accounts_bt_path = accounts_bt_file.path();
+    btree::BTreeIndex accounts_bt_index{accounts_bt_path.path()};
+
+    const auto accounts_kv_path_string{accounts_kv_path.path().string()};
+    const auto accounts_kvei_path_string{accounts_kvei_path.path().string()};
+    const auto accounts_bt_path_string{accounts_bt_path.path().string()};
+
+    // Prepare templates for C data structures of valid state (D/H/II) snapshots
+    SilkwormDomainSnapshot sample_domain_snapshot{
+        .segment = SilkwormMemoryMappedFile{
+            .file_path = accounts_kv_path_string.c_str(),
+            .memory_address = accounts_segment.memory_file_region().data(),
+            .memory_length = accounts_segment.memory_file_region().size(),
+        },
+        .existence_index = SilkwormMemoryMappedFile{
+            .file_path = accounts_kvei_path_string.c_str(),
+            .memory_address = nullptr,  // bloom filter is fully kept in memory, no mmap
+            .memory_length = 0,
+        },
+        .btree_index = SilkwormMemoryMappedFile{
+            .file_path = accounts_bt_path_string.c_str(),
+            .memory_address = accounts_bt_index.memory_file_region().data(),
+            .memory_length = accounts_bt_index.memory_file_region().size(),
+        },
+        .has_accessor_index = false,
+        .has_history = false,
+    };
+
+    // const snapshot_test::SampleAccountsDomainEFFile accounts_ef_file{tmp_dir.path()};
+    // const auto& accounts_ef_path = accounts_ef_file.path();
+
+    // const auto accounts_ef_path_string{accounts_ef_path.path().string()};
+
+    SilkwormInvertedIndexSnapshot sample_index_snapshot{
+        .segment = SilkwormMemoryMappedFile{
+            //.file_path = accounts_ef_path_string.c_str(),
+            //.memory_address = accounts_segment.memory_file_region().data(),
+            //.memory_length = accounts_segment.memory_file_region().size(),
+        },
+        .accessor_index = SilkwormMemoryMappedFile{
+            //.file_path = accounts_efi_path_string.c_str(),
+            //.memory_address = accounts_segment.memory_file_region().data(),
+            //.memory_length = accounts_segment.memory_file_region().size(),
```

Comment:
I suggest to either remove the commented code and unused empty SampleAccountsDomainEFFile class, or complete it. If you want to keep it and complete in the future, let's create a branch or a separate PR?